app/src/main/java/com/example/ncov19traking/ui/global/GlobalFragment.kt

```diff
@@ -94,9 +87,54 @@ class GlobalFragment : Fragment() {
         })
     }
 
-    override fun onAttach(context: Context) {
-        super.onAttach(context)
-        (requireActivity().applicationContext as BaseApp).applicationComponent.inject(this)
+    private fun GlobalFragmentViewHolder.receivingPercentageTextToCards(
+        nCoVInfo: NCoVInfo
+    ) {
+        covidTotalCasesCard.setPercentageText(
+            getString(
+                R.string.since_yesterday, homeViewModel.getCasesPercentageDifference(
+                    nCoVInfo.cases,
+                    homeViewModel.nCoVYesterdayAllCases.cases
+                )
+            )
+        )
+        covidRecoveredCard.setPercentageText(
+            getString(
+                R.string.since_yesterday, homeViewModel.getCasesPercentageDifference(
+                    nCoVInfo.recovered,
+                    homeViewModel.nCoVYesterdayAllCases.recovered
+                )
+            )
+        )
+        covidDeathsCard.setPercentageText(
+            getString(
+                R.string.since_yesterday, homeViewModel.getCasesPercentageDifference(
+                    nCoVInfo.deaths,
+                    homeViewModel.nCoVYesterdayAllCases.deaths
+                )
+            )
+        )
+    }
+
+    private fun GlobalFragmentViewHolder.setCovidDataToCards(
```

Comment:
this name could improve, you are only calling `setCaseNumbers` so I thing it should be better to reflect that in the method name, `data` is too broad It implies you are setting different kind of models inside this method.